### PR TITLE
Add Connection Documentation for Providers

### PR DESCRIPTION
**Pull request by sunkickr**
This PR documents connections for the following providers

- MySQL
- Slack
- Tableau
- Neo4j

### CODE REVIEW

File: airflow/providers/amazon/aws/transfers/mysql_to_s3.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class MySQLToS3Operator(BaseOperator):
     :type s3_bucket: str
     :param s3_key: desired key for the file. It includes the name of the file. (templated)
     :type s3_key: str
-    :param mysql_conn_id: reference to a specific mysql database
+    :param mysql_conn_id: :ref:Reference to `mysql connection id<howto/connection:mysql>`.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/transfers/mysql_to_s3.py]: Is this the correct quoting? I’d expect something like

```rst
:ref:`Reference to mysql connection id <howto/connection:mysql>`
```

or

```rst
Reference to :ref:`mysql connection id <howto/connection:mysql>`
```

Comment (sunkickr) [reply airflow/providers/amazon/aws/transfers/mysql_to_s3.py]: yes you right, sorry messed that up

Comment (sunkickr) [reply airflow/providers/amazon/aws/transfers/mysql_to_s3.py]: just fixed all the mistakes

**Comment by kaxil** [comment]
Please rebase on latest master please

**Comment by sunkickr** [comment]
@kaxil just rebased to the latest master thanks

**Comment by sunkickr** [comment]
There are few static errors I'll fix

**Review by kaxil — APPROVED** [review]

**Pull request merged**